fre/make/tests/null_example/test_checkout_null_yaml.py

```diff
@@ -0,0 +1,45 @@
+#tests for the create-checkout step of fre-make, for null_model.yaml
+import os
+from fre import fre
+from pathlib import Path
+from fre.pp import configure_script_yaml as csy
+from click.testing import CliRunner
+import subprocess
+from fre.make import createCheckout
+
+# Set example yaml paths, input directory
+test_dir = Path("fre/make/tests/null_example")
+yamlfile = Path(f"{test_dir}/null_model.yaml")
+
+#set platform and target
+platform = "ncrc5.intel"
+target = "debug"
+
+#set output directory
+#out_dir = Path(f"fre/make/tests/null_example/fre_make_out")
+#Path(out_dir).mkdir(parents=True,exist_ok=True)
+
+# Set home for ~/cylc-src location in script
+#os.environ["HOME"]=str(Path(f"{out_dir}"))
+HOME_DIR = os.environ["HOME"]
+#run checkout command
+runner = CliRunner()
+
+def test_checkout_script_exists():
+    """
+    Make sure checkout file exists
+    """
+    result = runner.invoke(fre.fre, args=["make","create-checkout","-y",yamlfile,"-p",platform,"-t",target])
```

Comment:
Using runner.invoke is good for testing the cli for fre. For actually script functionality testing here, we're going to call the actual functions in the script. 

Using `from fre.make import create_checkout_script` from the top:
1) we'll set/reset home as I mentioned above
2) we'll set what would be click options passed by the user for the fre make create-checkout command
3) we'll invoke the main checkout function 
4) check for the existence of the checkout script as a good starting test

You have line 33 that just needs to be edited a little. Try this:
`create_checkout_script.checkout_create(variables set right before)`